src/network/bmproto.py

```diff
@@ -277,25 +279,11 @@ def bm_command_error(self):
     def bm_command_getdata(self):
         items = self.decode_payload_content("l32s")
         # skip?
-        if time.time() < self.skipUntil:
+        now = time.time()
+        if now < self.skipUntil:
             return True
-        #TODO make this more asynchronous
-        helper_random.randomshuffle(items)
         for i in map(str, items):
```

Comment:
Why not
```python
for i in items:
    self.pendingUpload[str(i)] = now
```
?

Reply:
Leftover from old code, you're right, it can be rewritten for maybe a bit of performance increase.